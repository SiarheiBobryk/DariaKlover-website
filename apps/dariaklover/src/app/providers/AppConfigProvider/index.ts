export { default } from './AppConfigProvider';
export * from './AppConfigProvider';
