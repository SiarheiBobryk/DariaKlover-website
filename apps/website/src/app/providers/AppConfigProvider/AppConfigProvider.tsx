import * as React from 'react';

import { Config } from '@dariaklover/types';

export const config: Config = {
  fullName: 'Дарья Кловер',
  title: 'Дарья Кловер - Специалист и коуч по здоровью',
  url: 'https://dariaklover.health',
  description: 'Дарья Кловер - специалист и коуч по здоровью, проводник к здоровому образу жизни.',
  keywords:
    'health-коуч, коуч по здоровью, health coach, осознанность, здоровье, здоровое мышление, оздоровление, комплексный подход, холистический подход, физическая активность, сон, восстановление, адаптация, здоровая миграция, коучинг, саморазвитие, постановка целей, стресс менеджмент, уверенность в себе, проявленность, красота, питание, гармония, сопровождение, антиэйдж',
  language: 'ru',
  colorModeDefault: 'light', // The color-mode is managed by `ColorModeProvider`
  themeColor: {
    light: '#fff',
    dark: '#212121',
  },
  instagram: {
    id: '@daria.klover',
    href: 'https://www.instagram.com/daria.klover',
    label: 'Instagram Дарьи',
    labelShort: 'Instagram',
  },
  telegram: {
    id: 'dariaklover',
    href: 'https://t.me/dariaklover',
    label: 'Telegram Дарьи',
    labelShort: 'Telegram',
  },
  whatsapp: {
    // SOMEDAY: Setup the branded account id (payed feature) later
    id: '+46738540708',
    href: 'https://wa.me/+46738540708',
    label: 'WhatsApp Дарьи',
    labelShort: 'WhatsApp',
  },
  calendly: {
    id: 'dariaklover',
    href: 'https://calendly.com/dariaklover',
    label: 'Календарь Дарьи',
  },
  'msapplication-TileColor': '#ffc40d',
  email: {
    id: 'dariaklover@gmail.com',
    label: 'Электронная почта',
    labelShort: 'Email',
  },
  github: 'https://github.com/SiarheiBobryk/DariaKlover-website',
};

export const AppConfigContext = React.createContext<Config>(config);

export interface AppConfigProviderProps {
  children: React.ReactElement;
}

function AppConfigProvider(props: AppConfigProviderProps) {
  const { children } = props;
  return <AppConfigContext.Provider value={config}>{children}</AppConfigContext.Provider>;
}

export default AppConfigProvider;
