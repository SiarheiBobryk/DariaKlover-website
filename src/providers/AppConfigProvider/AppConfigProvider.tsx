import * as React from 'react';

import { PaletteMode } from '@mui/material';

export interface MetaTags {
  description: string;
  keywords: string;
  'msapplication-TileColor': string;
}

export interface CalendlyAppointment {
  id: string;
  active: boolean;
  newTab: boolean;
  label: string;
}

export interface Config extends MetaTags {
  fullName: string;
  title: string;
  url: string;
  language: string;
  colorModeDefault: PaletteMode;
  themeColor: {
    light: string;
    dark: string;
  };
  instagram: {
    id: string;
    href: string;
    label: string;
  };
  telegram: {
    id: string;
    href: string;
    label: string;
  };
  whatsapp: {
    id: string;
    href: string;
    label: string;
  };
  calendly: {
    id: string;
    href: string;
    label: string;
    appointments: Array<CalendlyAppointment>;
    conformation: boolean;
  };
  email: string;
  github: string;
}

export const config: Config = {
  fullName: 'Дарья Кловер',
  title: 'Дарья Кловер - Специалист и коуч по здоровью',
  url: 'https://dariaklover.health',
  description:
    // eslint-disable-next-line max-len
    'Образовательный сайт о здоровом образе жизни, инструментах и способах его достижения. Дарья Кловер - врач общей практики, косметик, блогер, зож-энтузиаст, будущий специалист и коуч по здоровью.',
  keywords: 'хелс коуч, health coach, здоровье, красота, питание, гармония, сопровождение, антиэйдж',
  language: 'ru',
  colorModeDefault: 'light', // NOTE: the color-mode is managed by `ColorModeProvider`
  themeColor: {
    light: '#fff',
    dark: '#212121',
  },
  instagram: {
    id: '@daria.klover',
    href: 'https://www.instagram.com/daria.klover',
    label: 'Инстаграм Дарьи',
  },
  telegram: {
    id: 'dariaklover',
    href: 'https://t.me/dariaklover',
    label: 'Телеграм Дарьи',
  },
  whatsapp: {
    // TODO: Setup the branded account id (payed feature) later
    id: '+46738540708',
    href: 'https://wa.me/+46738540708',
    label: 'WhatsApp Дарьи',
  },
  calendly: {
    id: 'dariaklover',
    href: 'https://calendly.com/dariaklover',
    label: 'Календарь Дарьи',
    appointments: [
      // NOTE: Due to limitations of a free account on Calendly, only one appointment type should be active
      { id: 'trial', active: true, newTab: true, label: 'Ознакомительная встреча' },
      { id: 'regular', active: false, newTab: true, label: 'Heath-коучинг сессия' },
    ],
    conformation: false,
  },
  'msapplication-TileColor': '#ffc40d',
  email: 'dariaklover@gmail.com',
  github: 'https://github.com/SiarheiBobryk/DariaKlover-website',
};

export const AppConfigContext = React.createContext<Config>(config);

export interface AppConfigProviderProps {
  children: React.ReactElement;
}

function AppConfigProvider(props: AppConfigProviderProps) {
  const { children } = props;
  return <AppConfigContext.Provider value={config}>{children}</AppConfigContext.Provider>;
}

export default AppConfigProvider;
