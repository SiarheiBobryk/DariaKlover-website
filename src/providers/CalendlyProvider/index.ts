export { default } from './CalendlyProvider';
export * from './CalendlyProvider';
