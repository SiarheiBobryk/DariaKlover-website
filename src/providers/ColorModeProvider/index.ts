export { default } from './ColorModeProvider';
export * from './ColorModeProvider';
