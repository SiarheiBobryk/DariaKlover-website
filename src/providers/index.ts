export { default as ColorModeProvider } from './ColorModeProvider';
export { default as ThemeProvider } from './ThemeProvider';
export { default as AppConfigProvider } from './AppConfigProvider';
