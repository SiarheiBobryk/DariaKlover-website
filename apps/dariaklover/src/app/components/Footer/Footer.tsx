import * as React from 'react';

import { Config } from '@dariaklover/types';
import CalendarMonthIcon from '@mui/icons-material/CalendarMonth';
import EmailIcon from '@mui/icons-material/Email';
import InstagramIcon from '@mui/icons-material/Instagram';
import TelegramIcon from '@mui/icons-material/Telegram';
import WhatsAppIcon from '@mui/icons-material/WhatsApp';
import Box from '@mui/material/Box';
import IconButton from '@mui/material/IconButton';
import Link from '@mui/material/Link';
import Tooltip from '@mui/material/Tooltip';
import Typography from '@mui/material/Typography';

import packageJson from '../../../../../../package.json';
import { AppConfigContext } from '../../providers/AppConfigProvider';

export const iconButtonA11yProps = {
  rel: 'noopener',
  target: '_blank',
};

interface FooterProps {}

function Footer(props: FooterProps) {
  const {
    instagram: { href: instagramHref, label: instagramLabel },
    telegram: { href: telegramHref, label: telegramLabel },
    whatsapp: { href: whatsappHref, label: whatsappLabel },
    calendly: { href: calendlyHref, label: calendlyLabel },
    fullName,
    github,
    email: { id: emailId },
  }: Config = React.useContext(AppConfigContext);

  return (
    <Box sx={{ display: 'flex', flexDirection: 'column', alignItems: 'center', gap: 1, marginBottom: 1 }} {...props}>
      <Box>
        <Tooltip title="Instagram">
          <IconButton aria-label={instagramLabel} href={instagramHref} color="inherit" {...iconButtonA11yProps}>
            <InstagramIcon />
          </IconButton>
        </Tooltip>
        <Tooltip title="Telegram">
          <IconButton aria-label={telegramLabel} href={telegramHref} color="inherit" {...iconButtonA11yProps}>
            <TelegramIcon />
          </IconButton>
        </Tooltip>
        <Tooltip title="WhatsApp">
          <IconButton aria-label={whatsappLabel} href={whatsappHref} color="inherit" {...iconButtonA11yProps}>
            <WhatsAppIcon />
          </IconButton>
        </Tooltip>
        <Tooltip title="Электронная почта">
          <IconButton
            aria-label="Электронный адрес Дарьи"
            href={`mailto:  ${emailId}`}
            color="inherit"
            {...iconButtonA11yProps}
          >
            <EmailIcon />
          </IconButton>
        </Tooltip>
        <Tooltip title="Календарь">
          <IconButton aria-label={calendlyLabel} href={calendlyHref} color="inherit" {...iconButtonA11yProps}>
            <CalendarMonthIcon />
          </IconButton>
        </Tooltip>
      </Box>
      <Typography variant="body2" color="text.secondary">
        {fullName} {new Date().getFullYear()}
      </Typography>
      <Tooltip title="Техническая информация о последнем обновлении">
        <Link variant="body2" color="text.secondary" href={`${github}/releases/tag/v${packageJson.version}`}>
          {`v${packageJson.version}`}
        </Link>
      </Tooltip>
    </Box>
  );
}

export default Footer;
