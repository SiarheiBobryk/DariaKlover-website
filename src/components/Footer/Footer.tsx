import * as React from 'react';

import Typography from '@mui/material/Typography';
import Link from '@mui/material/Link';
import IconButton from '@mui/material/IconButton';
import Box from '@mui/material/Box';
import InstagramIcon from '@mui/icons-material/Instagram';
import TelegramIcon from '@mui/icons-material/Telegram';
import CalendarMonthIcon from '@mui/icons-material/CalendarMonth';
import EmailIcon from '@mui/icons-material/Email';

import { AppConfigContext, Config } from '../../providers/AppConfigProvider';
import packageJson from '../../../package.json';

export const iconButtonA11yProps = {
  rel: 'noopener',
  target: '_blank',
};

interface FooterProps {
  // TODO: Add `data-testid` for all components
  'data-testid'?: string;
}

function Footer(props: FooterProps) {
  const {
    instagram: { href: instagramHref, label: instagramLabel },
    telegram: { href: telegramHref, label: telegramLabel },
    calendly: { href: calendlyHref, label: calendlyLabel },
    fullName,
    github,
    email,
  }: Config = React.useContext(AppConfigContext);

  return (
    <Box sx={{ display: 'flex', flexDirection: 'column', alignItems: 'center', gap: 1, marginBottom: 1 }} {...props}>
      <Box>
        <IconButton aria-label={instagramLabel} href={instagramHref} color="inherit" {...iconButtonA11yProps}>
          <InstagramIcon />
        </IconButton>
        <IconButton aria-label={telegramLabel} href={telegramHref} color="inherit" {...iconButtonA11yProps}>
          <TelegramIcon />
        </IconButton>
        <IconButton
          aria-label="Электронный адрес Дарьи"
          href={`mailto:  ${email}`}
          color="inherit"
          {...iconButtonA11yProps}
        >
          <EmailIcon />
        </IconButton>
        <IconButton aria-label={calendlyLabel} href={calendlyHref} color="inherit" {...iconButtonA11yProps}>
          <CalendarMonthIcon />
        </IconButton>
      </Box>
      <Typography variant="body2" color="text.secondary">
        {fullName} {new Date().getFullYear()}
      </Typography>
      <Link variant="body2" color="text.secondary" href={`${github}/releases/tag/v${packageJson.version}`}>
        {`v${packageJson.version}`}
      </Link>
    </Box>
  );
}

export default Footer;
