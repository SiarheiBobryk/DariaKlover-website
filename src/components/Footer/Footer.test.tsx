import * as React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Footer, { iconButtonA11yProps } from './Footer';
import { config } from '../../providers/AppConfigProvider';
import packageJson from '../../../package.json';

describe('<Footer />', () => {
  // TODO: Setup a11y testing by axe
  describe('snapshotting', () => {
    it('should render correctly', () => {
      // TODO: Add types for all test files
      const footer: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Footer />).toJSON();
      expect(footer).toMatchSnapshot();
    });
  });

  describe('Elements inspection', () => {
    it('should have the right instagram icon button', () => {
      render(<Footer />);
      const a: HTMLElement | null = document.querySelector(`a[href="${config.instagram.href}"]`);
      expect(a).toHaveAttribute('rel', iconButtonA11yProps.rel);
      expect(a).toHaveAttribute('target', iconButtonA11yProps.target);
      expect(a).toHaveAttribute('aria-label', config.instagram.label);

      const svg: ChildNode | null | undefined = a?.firstChild;
      expect(svg).toHaveAttribute('data-testid', 'InstagramIcon');
    });

    it('should have the right telegram icon button', () => {
      render(<Footer />);
      const a: HTMLElement | null = document.querySelector(`a[href="${config.telegram.href}"]`);
      expect(a).toHaveAttribute('rel', iconButtonA11yProps.rel);
      expect(a).toHaveAttribute('target', iconButtonA11yProps.target);
      expect(a).toHaveAttribute('aria-label', config.telegram.label);

      const svg: ChildNode | null | undefined = a?.firstChild;
      expect(svg).toHaveAttribute('data-testid', 'TelegramIcon');
    });

    it('should render name and year', () => {
      render(<Footer />);
      const p: HTMLElement = screen.getByText(`${config.fullName} ${new Date().getFullYear()}`);
      expect(p).toBeInTheDocument();
    });

    it('should render the application version', () => {
      render(<Footer />);
      const p: HTMLElement = screen.getByText(`v.${packageJson.version}`);
      expect(p).toBeInTheDocument();
    });
  });

  describe('Styling inspection', () => {
    it('should be centered, direction column and have a gap', () => {
      render(<Footer data-testid="footer" />);
      const footer: HTMLElement = screen.getByTestId('footer');
      expect(footer).toHaveStyle({
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        gap: '8px',
      });
    });
  });
});
