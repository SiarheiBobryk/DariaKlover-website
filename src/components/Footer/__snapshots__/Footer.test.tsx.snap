// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> snapshotting should render correctly 1`] = `
<div
  className="MuiBox-root css-s73z7h"
>
  <div
    className="MuiBox-root css-0"
  >
    <a
      aria-label="Instagram Дарьи"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element={true}
      href="https://www.instagram.com/daria.klover"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      rel="noopener"
      tabIndex={0}
      target="_blank"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="InstagramIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
    <a
      aria-label="Telegram Дарьи"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element={true}
      href="https://t.me/dariaklover"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      rel="noopener"
      tabIndex={0}
      target="_blank"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="TelegramIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M9.78 18.65l.28-4.23 7.68-6.92c.34-.31-.07-.46-.52-.19L7.74 13.3 3.64 12c-.88-.25-.89-.86.2-1.3l15.97-6.16c.73-.33 1.43.18 1.15 1.3l-2.72 12.81c-.19.91-.74 1.13-1.5.71L12.6 16.3l-1.99 1.93c-.23.23-.42.42-.83.42z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
    <a
      aria-label="WhatsApp Дарьи"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element={true}
      href="https://wa.me/+375447841051"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      rel="noopener"
      tabIndex={0}
      target="_blank"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="WhatsAppIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.75 13.96c.25.13.41.2.46.3.06.11.04.61-.21 1.18-.2.56-1.24 1.1-1.7 1.12-.46.02-.47.36-2.96-.73-2.49-1.09-3.99-3.75-4.11-3.92-.12-.17-.96-1.38-.92-2.61.05-1.22.69-1.8.95-2.04.24-.26.51-.29.68-.26h.47c.15 0 .36-.06.55.45l.69 1.87c.06.13.1.28.01.44l-.27.41-.39.42c-.12.12-.26.25-.12.5.12.26.62 1.09 1.32 1.78.91.88 1.71 1.17 1.95 1.3.24.14.39.12.54-.04l.81-.94c.19-.25.35-.19.58-.11l1.67.88M12 2a10 10 0 0 1 10 10 10 10 0 0 1-10 10c-1.97 0-3.8-.57-5.35-1.55L2 22l1.55-4.65A9.969 9.969 0 0 1 2 12 10 10 0 0 1 12 2m0 2a8 8 0 0 0-8 8c0 1.72.54 3.31 1.46 4.61L4.5 19.5l2.89-.96A7.95 7.95 0 0 0 12 20a8 8 0 0 0 8-8 8 8 0 0 0-8-8z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
    <a
      aria-label="Электронный адрес Дарьи"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element={true}
      href="mailto:  dariaklover@gmail.com"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      rel="noopener"
      tabIndex={0}
      target="_blank"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="EmailIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
    <a
      aria-label="Календарь Дарьи"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element={true}
      href="https://calendly.com/dariaklover"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      rel="noopener"
      tabIndex={0}
      target="_blank"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CalendarMonthIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
  >
    Дарья Кловер
     
    2022
  </p>
  <a
    aria-label="Техническая информация о последнем обновлении"
    aria-labelledby={null}
    className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-1x1e5md-MuiTypography-root-MuiLink-root"
    data-mui-internal-clone-element={true}
    href="https://github.com/SiarheiBobryk/DariaKlover-website/releases/tag/v0.0.26"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
  >
    v0.0.26
  </a>
</div>
`;
